### PR TITLE
make the recycler pod template changeable

### PR DESCRIPTION
**Pull request by deads2k**
Fixes https://github.com/openshift/origin/issues/15812

### CODE REVIEW

File: pkg/cmd/server/kubernetes/master/controller/volumes.go [hunk 1]
```diff
@@ -1,128 +1,29 @@
...
-	nfsConfig := volume.VolumeConfig{
-		RecyclerMinimumTimeout:   int(config.PersistentVolumeRecyclerConfiguration.MinimumTimeoutNFS),
-		RecyclerTimeoutIncrement: int(config.PersistentVolumeRecyclerConfiguration.IncrementTimeoutNFS),
-		RecyclerPodTemplate:      defaultScrubPod,
+// newPersistentVolumeRecyclerPodTemplate provides a function which makes our recycler pod template for use in the kube-controller-manager
+// this is a stop-gap until the kube-controller-manager take a pod manifest
+func newPersistentVolumeRecyclerPodTemplate(recyclerImageName string) func() *v1.Pod {
+	return func() *v1.Pod {
+		uid := int64(0)
+		defaultScrubPod := volume.NewPersistentVolumeRecyclerPodTemplate()
```
Comment (jsafrane) [inline pkg/cmd/server/kubernetes/master/controller/volumes.go]: wouldn't it recursively call `newPersistentVolumeRecyclerPodTemplate()` (the one that starts 3 lines above)?  IMO you need proper closure:

```go
func newPersistentVolumeRecyclerPodTemplate(...) {
    oldTemplateFunc := volume.NewPersistentVolumeRecyclerPodTemplate()
    return func() *v1.Pod {
		uid := int64(0)
		defaultScrubPod := oldTemplateFunc()
...
```

I am curious of test results...

Comment (deads2k) [inline pkg/cmd/server/kubernetes/master/controller/volumes.go]: > I am curious of test results...

Heh, fully red.

Comment (deads2k) [inline pkg/cmd/server/kubernetes/master/controller/volumes.go]: fixed

**Comment by jsafrane** [comment]
/lgtm
/retest

flake: #15961

**Pull request merged**